servicebroker/servicebroker.go

```diff
@@ -184,32 +160,118 @@ func (broker *ServiceBroker) Services(ctx context.Context) []brokerapi.Service {
+func (broker *ServiceBroker) Provision(
+	ctx context.Context,
+	instanceID string,
+	details brokerapi.ProvisionDetails,
+	asyncAllowed bool,
+) (brokerapi.ProvisionedServiceSpec, error) {
+	spec := brokerapi.ProvisionedServiceSpec{}
+
+	err := broker.router.VerifyPlanExists(details.PlanID)
+	if err != nil {
+		broker.logger.Warn("broker-provision-plan-error", zap.Error(err))
+	}
+
+	return spec, err
 }
 
-// Deprovision service
-func (broker *ServiceBroker) Deprovision(ctx context.Context, instanceID string, details brokerapi.DeprovisionDetails, asyncAllowed bool) (brokerapi.DeprovisionServiceSpec, error) {
+// Deprovision stubbed since the bigip service will not be deprovisioned for now
+func (broker *ServiceBroker) Deprovision(
+	ctx context.Context,
+	instanceID string,
+	details brokerapi.DeprovisionDetails,
+	asyncAllowed bool,
+) (brokerapi.DeprovisionServiceSpec, error) {
 	return brokerapi.DeprovisionServiceSpec{}, nil
 }
 
-// Bind instance to service
-func (broker *ServiceBroker) Bind(ctx context.Context, instanceID string, bindingID string, details brokerapi.BindDetails) (brokerapi.Binding, error) {
-	return brokerapi.Binding{}, nil
+// Bind apply bigip configurations to bigip objects based on requested bind route
+func (broker *ServiceBroker) Bind(
+	ctx context.Context,
+	instanceID string,
+	bindID string,
+	details brokerapi.BindDetails,
+) (brokerapi.Binding, error) {
+	var err error
+	apiErr := false
+	binding := brokerapi.Binding{}
+
+	if details.BindResource == nil {
+		apiErr = true
+	} else if details.BindResource.Route == "" {
+		apiErr = true
+	}
+
+	if apiErr {
+		err = errors.New("only route services bindings are supported")
+		broker.logger.Warn("broker-route-bind-error", zap.Error(err))
+		return binding, err
+	}
+
+	err = broker.router.VerifyPlanExists(details.PlanID)
+	if err != nil {
+		broker.logger.Warn("broker-bind-plan-error", zap.Error(err))
+		return binding, err
+	}
+
+	// Map bindID to route URI then create new route update to give to router
+	routeURI := details.BindResource.Route
+	broker.bindIDRouteMap[bindID] = routeURI
+	ru, err := f5router.NewBrokerUpdate(broker.logger, routeUpdate.Bind, route.Uri(routeURI), details.PlanID)
+	if err != nil {
+		broker.logger.Warn("broker-bind-route-update-error", zap.Error(err))
+		return binding, err
+	}
+	broker.router.UpdateRoute(ru)
+
+	return binding, nil
 }
 
-// Unbind instance from service
-func (broker *ServiceBroker) Unbind(ctx context.Context, instanceID string, bindingID string, details brokerapi.UnbindDetails) error {
+// Unbind remove applied bigip configurations and bigip objects based on requested unbind route
+func (broker *ServiceBroker) Unbind(
+	ctx context.Context,
+	instanceID string,
+	bindID string,
+	details brokerapi.UnbindDetails,
+) error {
+	var err error
+	routeURI := broker.bindIDRouteMap[bindID]
+
+	if routeURI == "" {
+		err = errors.New("unbind route not found")
+		broker.logger.Warn("broker-unbind-plan-error", zap.Error(err))
```

Comment:
remove logs on errors from bind and unbinds

Reply:
caveat do these show up in the lager logger with the cf-broker package

Reply:
Found a lager adapter for our logger so I am now passing that in as the cf-service-broker logger. So these logs have been removed and errors returned only.